src/FilenameParsing/ParsedFileID.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace SilverStripe\Assets\FilenameParsing;
+
+/**
+ * Immutable representation of a parsed fileID broken down into its sub-components.
+ */
+class ParsedFileID
```

Comment:
This class is really just another drop-in alternative to the tuple array right? Do we need both? Do we have a clear delineation for when one is used over another?

Reply:
In a perfect world, we would only use ParsedFileID. However, right now we'll consider it an `@internal` class. It's only being used when interacting with a FileResolutionStrategy at this time.

Reply:
https://github.com/silverstripe/silverstripe-assets/pull/223#issuecomment-484260132

Reply:
Happy with the `@internal`